spmd/distributed.py

```diff
@@ -0,0 +1,94 @@
+import io
+from dataclasses import replace
+from typing import Any
+
+import torch
+import torch.distributed.distributed_c10d as c10d
+from torch.distributed import ProcessGroup
+from torch.distributed._shard.checkpoint.default_planner import (
+    DefaultLoadPlanner,
+    DefaultSavePlanner,
```

Comment:
I was trying to understand these planner classes, but failed to find them in https://github.com/pytorch/pytorch/blob/master/torch/distributed/_shard/checkpoint/__init__.py
Are these classes defined somewhere else?

Reply:
They are in the process of being merged. This change requires a mashup of multiple PRs (dist.cp, fsdp+tp integration) plus fixes on top to them to work.

At this point point, this is the best place to land this change as all work around DistributedTensor is converging here.